src/dkg/round1.rs

```diff
@@ -159,10 +275,18 @@ where
     let encrypted_secret_package =
         export_secret_package(&secret_package, self_identity, &mut csrng)
             .map_err(Error::EncryptionError)?;
-    let public_package = public_package.serialize().map_err(Error::FrostError)?;
 
-    // TODO bind the min/max signers and the list of participants to the packages through
-    // checksumming
+    let group_secret_key_shard: [u8; 32] = GroupSecretKeyShard::random(&mut csrng).serialize();
+
+    let public_package = PublicPackage::new(
+        self_identity.clone(),
+        min_signers,
+        &participants,
+        public_package,
+        group_secret_key_shard,
+    );
+    let public_package = public_package.serialize();
```

Comment:
I would return the `PublicPackage` directly to the caller, without serializing it.

The `SecretPackage` is returned as bytes because it's encrypted, and I wanted to save a few lines to the caller, but for the `PublicPackage` we can avoid returning bytes to allow greater flexibility.

Reply:
Updated!